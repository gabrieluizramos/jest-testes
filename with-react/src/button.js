import React from 'react';

export const Button = ({ onClick, children }) => {
    const link = <a href='google'>Link</a>;

    return (
        <>
            <button onClick={onClick} className="botao-reutilizavel">
                {children}
            </button>
            {/* {link} */}
        </>
    )
} 