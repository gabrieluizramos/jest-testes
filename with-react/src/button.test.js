import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { Button } from './button';

describe('Button', () => {
    it('renders with childrens', () => {
        const { asFragment } = render(<Button>qualquerConteudo</Button>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('clicks', () => {
        const onClick = jest.fn();
        render(<Button onClick={onClick}>qualquerConteudo</Button>);

        const button = screen.getByRole('button');
        fireEvent.click(button);

        expect(onClick).toHaveBeenCalledTimes(1);
    });
});

