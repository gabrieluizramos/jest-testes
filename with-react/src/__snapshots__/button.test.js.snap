// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders with childrens 1`] = `
<DocumentFragment>
  <button
    class="botao-reutilizavel"
  >
    qualquerConteudo
  </button>
</DocumentFragment>
`;
